ibis/expr/sql.py

```diff
@@ -336,6 +336,23 @@ def __repr__(self) -> str:
     def _repr_markdown_(self) -> str:
         return f"```sql\n{self!s}\n```"
 
+    def _repr_pretty_(self, p, cycle) -> str:
+        try:
+            from pygments import highlight
+            from pygments.formatters import TerminalFormatter
+            from pygments.lexers import SqlLexer
+        except ImportError:
+            output = str(self)
+        else:
+            output = highlight(
```

Comment:
Is there a chance that highlighting can error for any of the dialects we support?

Also, these lines appear uncovered, do we want to add a test for this?

Reply:
Possibly, but it's hard to say.

Picking an arbitrary example not common to most sql dialects (duckdb's struct literal syntax), highlighting seems to work:

![image](https://github.com/ibis-project/ibis/assets/417981/944d67dd-6923-4a59-8ffd-076f30c5e0a1)

I can catch `Exception` and warn in the case anything goes wrong (or maybe just avoid warning?)

Reply:
Added a `contextlib.suppress` around the `highlight` call.